packages/amplify_storage_s3/lib/src/S3List/S3ListOptions.dart

```diff
@@ -11,11 +11,12 @@ class S3ListOptions extends ListOptions {
 
   @override
   Map<String, dynamic> serializeAsMap() {
-    final Map<String, dynamic> optionsMap = {
-      'accessLevel': describeEnum(accessLevel),
-      'targetIdentityId': targetIdentityId
-    };
-    optionsMap.removeWhere((_, v) => v == null);
+    final Map<String, dynamic> optionsMap = <String, dynamic>{};
+    optionsMap["accessLevel"] = describeEnum(accessLevel);
+
+    if (targetIdentityId != null) {
+      optionsMap["targetIdentityId"] = targetIdentityId;
+    }
```

Comment:
Nit: We can still initialize this with accessLevel defined in the map literal right?
```dart
final Map<String, dynamic> optionsMap = <String, dynamic>{
  'accessLevel' = describeEnum(accessLevel)
};
```
I may have missed something in a different PR but I'm curious - why move away from the `removeWhere` pattern in just this one particular case?

Reply:
Yes we can initialize in the map literal, I think this was a merge conflict override so I switched it back: 

` optionsMap["accessLevel"] = {describeEnum(accessLevel)};
`

Reply:
Any concerns with switching back to using
```dart
optionsMap.removeWhere((_, v) => v == null);
```
to stay consistent?

Reply:
Yeah good point, should probably do that.